Sources/AsyncHTTPClient/ConnectionPool/HTTP1.1/HTTP1ClientChannelHandler.swift

```diff
@@ -0,0 +1,347 @@
+
+        case .fireChannelActive:
+            context.fireChannelActive()
+
+        case .fireChannelInactive:
+            context.fireChannelInactive()
+
+        case .fireChannelError(let error, let close):
+            context.fireErrorCaught(error)
+            if close {
+                context.close(promise: nil)
+            }
+
+        case .read:
+            context.read()
+
+        case .close:
+            context.close(promise: nil)
+
+        case .wait:
+            break
+
+        case .forwardResponseHead(let head, let pauseRequestBodyStream):
+            self.request!.receiveResponseHead(head)
+            if pauseRequestBodyStream {
+                self.request!.pauseRequestBodyStream()
+            }
+
+        case .forwardResponseBodyParts(let buffer):
+            self.request!.receiveResponseBodyParts(buffer)
+
+        case .succeedRequest(let finalAction, let buffer):
+            // The order here is very important...
+            // We first nil our own task property! `taskCompleted` will potentially lead to
+            // situations in which we get a new request right away. We should finish the task
+            // after the connection was notified, that we finished. A
+            // `HTTPClient.shutdown(requiresCleanShutdown: true)` will fail if we do it the
+            // other way around.
+
+            let oldRequest = self.request!
+            self.request = nil
+
+            switch finalAction {
+            case .close:
+                context.close(promise: nil)
+            case .sendRequestEnd:
+                context.writeAndFlush(self.wrapOutboundOut(.end(nil)), promise: nil)
+            case .informConnectionIsIdle:
+                self.connection.taskCompleted()
+            case .none:
+                break
+            }
+
+            oldRequest.succeedRequest(buffer)
+
+        case .failRequest(let error, let finalAction):
+            let oldRequest = self.request!
+            self.request = nil
+
+            switch finalAction {
+            case .close:
+                context.close(promise: nil)
+            case .sendRequestEnd:
+                context.writeAndFlush(self.wrapOutboundOut(.end(nil)), promise: nil)
+            case .informConnectionIsIdle:
+                self.connection.taskCompleted()
+            case .none:
+                break
+            }
+
+            oldRequest.fail(error)
+        }
+    }
+
+    // MARK: - Private Methods -
+
+    private func resetIdleReadTimeoutTimer(_ idleReadTimeout: TimeAmount, context: ChannelHandlerContext) {
+        if let oldTimer = self.idleReadTimeoutTimer {
+            oldTimer.cancel()
+        }
+
+        self.idleReadTimeoutTimer = context.channel.eventLoop.scheduleTask(in: idleReadTimeout) {
+            let action = self.state.idleReadTimeoutTriggered()
+            self.run(action, context: context)
+        }
+    }
+
+    private func clearIdleReadTimeoutTimer() {
```

Comment:
Is this ever called? Looks like it was superseded by `resetIdleReadTimeoutTimer`.

Reply:
Added code that uses that :)